cg/meta/report/raredisease.py

```diff
@@ -0,0 +1,131 @@
+import logging
+
+from cg.constants import (
+    REQUIRED_REPORT_FIELDS,
+    REQUIRED_CUSTOMER_FIELDS,
+    REQUIRED_CASE_FIELDS,
+    REQUIRED_APPLICATION_FIELDS,
+    REQUIRED_SAMPLE_RAREDISEASE_FIELDS,
+    REQUIRED_DATA_ANALYSIS_RAREDISEASE_FIELDS,
+    REQUIRED_SAMPLE_METHODS_FIELDS,
+    REQUIRED_SAMPLE_TIMESTAMP_FIELDS,
+    REQUIRED_SAMPLE_METADATA_RAREDISEASE_WGS_FIELDS,
+    REQUIRED_SAMPLE_METADATA_RAREDISEASE_FIELDS,
+)
+from cg.constants.scout import RAREDISEASE_CASE_TAGS
+from cg.meta.report.field_validators import get_million_read_pairs
+from cg.meta.report.report_api import ReportAPI
+from cg.models.analysis import NextflowAnalysis, AnalysisModel
+from cg.models.cg_config import CGConfig
+from cg.meta.workflow.raredisease import RarediseaseAnalysisAPI
+from cg.models.raredisease.raredisease import RarediseaseQCMetrics
+from cg.models.report.metadata import RarediseaseSampleMetadataModel
+from cg.models.report.report import ScoutReportFiles, CaseModel
+from cg.models.report.sample import SampleModel
+from cg.store.models import Case, Sample
+
+LOG = logging.getLogger(__name__)
+
+
+class RarediseaseReportAPI(ReportAPI):
+    """API to create Rare disease DNA delivery reports."""
+
+    def __init__(self, config: CGConfig, analysis_api: RarediseaseAnalysisAPI):
+        super().__init__(config=config, analysis_api=analysis_api)
+
+    def get_sample_metadata(
+        self, case: Case, sample: Sample, analysis_metadata: NextflowAnalysis
+    ) -> RarediseaseSampleMetadataModel:
+        """Return sample metadata to include in the report."""
+        sample_metrics: RarediseaseQCMetrics = analysis_metadata.sample_metrics[sample.internal_id]
+
+        sample_coverage: dict = self.get_sample_coverage(sample=sample, case=case)
+        return RarediseaseSampleMetadataModel(
+            bait_set=self.lims_api.capture_kit(lims_id=sample.internal_id),
+            gender=sample_metrics.predicted_sex,
+            million_read_pairs=get_million_read_pairs(reads=sample.reads),
+            mapped_reads=sample_metrics.percentage_mapped_reads,
+            mean_target_coverage=sample_coverage.get("mean_coverage"),
+            pct_10x=sample_coverage.get("mean_completeness"),
+            duplicates=sample_metrics.fraction_duplicates,
+        )
+
+    ##TODO #here it looks for chanjo API, need to add chanjo2 instead
+    def get_sample_coverage(self, sample: Sample, case: Case) -> dict:
+        """Return coverage values for a specific sample."""
+        genes = self.get_genes_from_scout(panels=case.panels)
```

Comment:
The api endpoint in chanjo2 that you are calling is expecting a bed file with the gene coordinates, not gene IDs. So perhaps export the gene panel from scout as a bed file? 

In scout you would do it with this command: `scout export panel --build 37 --bed OMIM-AUTO` but I don't know if there is some shortcut in cg to do the same. Then I would save this bed file as a temporary file and send it as a request to the chanjo2 api (intervals_bed_path param)